packages/smart-contracts/test/contracts/EthereumProxy.test.ts

```diff
@@ -37,20 +40,18 @@ describe('contract: EthereumProxy', () => {
     const fromOldBalance = await provider.getBalance(from);
     const toOldBalance = await provider.getBalance(to);
 
-    const receipt = await (
+    await (
       await ethProxy.transferWithReference(to, referenceExample, {
         value: amount,
         gasPrice: DEFAULT_GAS_PRICE,
       })
     ).wait();
 
-    const gasCost = DEFAULT_GAS_PRICE.mul(BigNumber.from(receipt.gasUsed));
-
     const fromNewBalance = await provider.getBalance(from);
     const toNewBalance = await provider.getBalance(to);
 
     // Check balance changes
-    expect(fromNewBalance.toString()).to.equals(fromOldBalance.sub(gasCost).sub(amount).toString());
+    expect(fromNewBalance).to.be.lt(fromOldBalance.sub(amount));
```

Comment:
the previous check seemed more precise, what's the limitation?

Reply:
The limitation is that on L2 network the formula to compute gas fees is not the same as on mainnet. And neither it is between two L2 networks.

Reply:
ok, but the receipt doesn't contain the final value? this is very weird 🤔

Reply:
Not always. 

On optimism for example you can compute the L2 fees (cost of computing the tx on Optimism) from the receipt.
But then you also need to compute the L1 fees (cost of storing the tx on Ethereum).
This is not done through the receipt (an oracle has to be called according to the doc - https://community.optimism.io/docs/developers/build/transaction-fees/#displaying-fees-to-users).

Reply:
hum I see. OK , never mind then